app/src/main/java/nl/tudelft/b_b_w/model/block/GenesisBlock.java

```diff
@@ -26,12 +29,39 @@
     public GenesisBlock(BlockData blockData) throws HashException {
         super(blockData);
 
-        // verify integration and integrity
-        if (BuildConfig.DEBUG && blockData.getBlockType() != BlockType.GENESIS
-                || blockData.getSequenceNumber() != firstSequenceNumber
-                || !blockData.getPreviousHashChain().equals(notAvailable)
-                || !blockData.getPreviousHashSender().equals(notAvailable)) {
-            throw new AssertionError("Invalid genesis block " + this.toString());
+        // when running in debug mode, verify that all properties are set correctly
+        if (BuildConfig.DEBUG) {
+            boolean wrongType = blockData.getBlockType() != BlockType.GENESIS;
```

Comment:
I think that Jasper also has to explain his exercise; a constructor can contain some code and since it is the only code this constructor contains I think it is reasonable